h2o/h2o
# Mtls basic feature

Implement basic feature for mutual TLS (client authentication), both TLS 1.2 and 1.3
-Add 1 optional configuration to enable client certificate verification, for example,
    _client-ca-file: examples/h2o/test_client_ca.crt_
The option requires a proper client CA root file or bundle. The client certificate authentication validates standard cert chain signature and expiration.

CI test:
40mtls.t and 50reverse-proxy-mtls.t tests are included.

Local manual test:
1. add client-ca-file option in h2o configuration file, use examples/h2o/test_client_ca.crt 
2. start ./h2o, make sure relative path is OK so that all cert, key and root are loaded.
In another terminal/console:
go to  h2o home folder
run "curl -v --cacert misc/test-ca/ca.crt --key t/assets/test_client.key --cert t/assets/test_client.crt  https://127.0.0.1.xip.io:8081"
